src/hot_topic.cr

```diff
@@ -30,5 +30,65 @@ module HotTopic
 
       HTTP::Client::Response.from_io(buffer.rewind)
     end
+
+    def establish_ws(uri : URI | String, headers = HTTP::Headers.new) : HTTP::WebSocket
+      # build bi-directional io
+      local_read, remote_write = IO.pipe
+      remote_read, local_write = IO.pipe
+      local_io = IO::Stapled.new(local_read, local_write)
+      remote_io = IO::Stapled.new(remote_read, remote_write)
```

Comment:
When you posted #1 I started looking into it and came up with a pretty similar approach here. 😄 

<img width="684" alt="Screenshot of code similar to the code above" src="https://user-images.githubusercontent.com/108205/170804723-caceb47d-654f-40e3-b1c2-e9d02158348e.png">